caching_of_git_repos/README.md

```diff
@@ -0,0 +1,102 @@
+# How to cache git repositories with a reference repo
+
+## The main idea
+
+Using `git clone --reference /repo/with/cache https://some/repo/to/clone`.
+
+What it does?
+
+- When cloning a repo, you can specify a repo, that will be searched first for commits to download.
+  (If the commit is found in the reference repo, reference is used instead of saving the data.
+  If not found, commit is downloaded and data saved.)
+
+  --reference[-if-able] <repository>
+  If the reference repository is on the local machine, automatically setup .git/objects/info/alternates to obtain objects from the reference repository. Using an already existing repository as an alternate will require
+  fewer objects to be copied from the repository being cloned, reducing network and local storage costs. When using the --reference-if-able, a non existing directory is skipped with a warning instead of aborting the clone.
+
+- The state of the reference repo is not important.
+  (We don't need to care about force-pushes, merging, local changes,...)
+  What is fetched counts.
+- We can fetch commits of multiple repositories in the reference repository.
+  - Alternatively, we can use more cache repositories:
+    `git clone --reference-if-able /cache/some-repo-to-clone https://some/repo/to/clone`
+
+## Pros&Cons
+
+- Cloning is much faster for repositories in the cache.
+- Cloning is slower for repositories not present in the cache.
+- Less memory is needed to clone repositories in the cache.
+  (Which makes it possible to clone kernel for example.)
+- More memory is needed to clone repositories not present in the cache.
```

Comment:
it depends what would be the workflow to populate the cache but we should do it outside sandcastle/worker